packages/util-git-info/examples/serverless-monorepo-detection.js

```diff
@@ -0,0 +1,319 @@
+        )
+        const changeIndicator = isChanged ? '🔄' : '  '
+        console.log(`      ${changeIndicator} ${func.name}: ${func.handler}`)
+        if (isChanged) {
+          console.log(`         File: ${func.file}`)
+        }
+      })
+    }
+
+    console.log()
+  })
+
+  console.log('═══════════════════════════════════════════')
+  console.log('\n🔧 Recommended Actions:\n')
+
+  changedProjects.forEach(project => {
+    console.log(`   📦 ${project.name}:`)
+
+    if (project.packageJsonChanged) {
+      console.log(`      • Run: cd ${project.path} && npm install`)
+    }
+
+    if (project.handlerChanges.length > 0) {
+      console.log(`      • Test ${project.handlerChanges.length} changed function(s)`)
+      console.log(`      • Run: cd ${project.path} && serverless deploy`)
+    }
+
+    if (project.configChanged) {
+      console.log(`      • Review serverless configuration changes`)
+      console.log(`      • Validate: cd ${project.path} && serverless package`)
+    }
+
+    console.log()
+  })
+
+  // Export for CI/CD pipelines
+  console.log('📋 Changed projects (JSON for CI/CD):')
+  console.log(JSON.stringify(changedProjects.map(p => ({
+    name: p.name,
+    path: p.path,
+    packageJsonChanged: p.packageJsonChanged,
+    configChanged: p.configChanged,
+    functionsChanged: p.handlerChanges.length,
+    handlers: p.functionHandlers.map(h => h.handler)
+  })), null, 2))
+}
+
+/**
+ * Find all directories containing serverless configuration files
+ */
+async function findServerlessDirectories(repoDir) {
+  const serverlessConfigPatterns = [
+    'serverless.yml',
+    'serverless.yaml',
+    'serverless.ts',
+    'serverless.js'
+  ]
+
+  const directories = new Set()
+
+  function searchDirectory(dir) {
+    try {
+      const entries = fs.readdirSync(dir, { withFileTypes: true })
+
+      for (const entry of entries) {
+        const fullPath = path.join(dir, entry.name)
+
+        // Skip node_modules and hidden directories
+        if (entry.name === 'node_modules' || entry.name.startsWith('.')) {
+          continue
+        }
+
+        if (entry.isDirectory()) {
+          searchDirectory(fullPath)
+        } else if (entry.isFile() && serverlessConfigPatterns.includes(entry.name)) {
+          directories.add(dir)
+        }
+      }
+    } catch (err) {
+      // Skip directories we can't read
+    }
+  }
+
+  searchDirectory(repoDir)
+  return Array.from(directories)
+}
+
+/**
+ * Get the serverless config file path for a directory
+ */
+function getServerlessConfigFile(dir) {
+  const configFiles = [
+    'serverless.yml',
+    'serverless.yaml',
+    'serverless.ts',
+    'serverless.js'
+  ]
+
+  for (const file of configFiles) {
+    const filePath = path.join(dir, file)
+    if (fs.existsSync(filePath)) {
+      return filePath
+    }
+  }
+
+  return null
+}
+
+/**
+ * Extract function handlers from serverless configuration file
+ */
+async function extractFunctionHandlers(configFile) {

```

Comment:
change the extract handler to use the configorama npm package it will parse whatever flavor of config and give us a usable object

https://github.com/DavidWells/configorama